editor/editor_node.cpp

```diff
@@ -1811,11 +1811,21 @@ void EditorNode::save_all_scenes() {
 	_save_all_scenes();
 }
 
-void EditorNode::save_scene_list(Vector<String> p_scene_filenames) {
+void EditorNode::save_scene_if_open(const String &p_scene_path) {
 	for (int i = 0; i < editor_data.get_edited_scene_count(); i++) {
 		Node *scene = editor_data.get_edited_scene_root(i);
 
-		if (scene && (p_scene_filenames.find(scene->get_scene_file_path()) >= 0)) {
+		if (scene && scene->get_scene_file_path() == p_scene_path) {
+			_save_scene(scene->get_scene_file_path(), i);
```

Comment:
```suggestion
			_save_scene(scene->get_scene_file_path(), i);
			return;
```
I don't believe that two scenes can have the same path

Reply:
Or better yet, replace the whole code with:
```cpp
int idx = editor_data.get_edited_scene_from_path(p_scene_path);
if (idx >= 0) {
	_save_scene(p_scene_path, idx);
}
```
Unless I'm mistaken

Reply:
That's a convenient function, thanks

Reply:
Might need to check that the scene is valid as well as in the original, but try it out